File: SWEA/D3/1289. 원재의 메모리 복구하기/원재의 메모리 복구하기.py
T = int(input())

for tc in range(1, T + 1):
    bit = input()
    bits = []
    for b in bit:
        bits.append(int(b))

    answer = [0 for _ in range(len(bits))]
    count = 0
    i = 0

    while True:
        if bits == answer:
            break
        if bits[i] != answer[i]:
            for j in range(i, len(bits)):
                if bits[j] == 0:
                    bits[j] = 1
                else:
                    bits[j] = 0
            count += 1
        i += 1
    print(f'#{tc} {count}')

File: SWEA/D3/1289. 원재의 메모리 복구하기/원재의 메모리 복구하기.java
import java.io.*;
import java.util.*;

public class Solution {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st;

        st = new StringTokenizer(br.readLine());
        int T = Integer.parseInt(st.nextToken());
        for (int t = 1; t <= T; t++) {
            System.out.print("#" + t + " ");

            st = new StringTokenizer(br.readLine());
            String bit = st.nextToken();
            char current = '0';
            int count = 0;

            for (int i = 0; i < bit.length(); i++) {
                if (bit.charAt(i) != current) {
                    count += 1;
                    current = bit.charAt(i);
                }
            }
            System.out.println(count);
        }
    }
}